gobblin-core/src/main/java/org/apache/gobblin/writer/FsDataWriter.java

```diff
@@ -247,6 +256,18 @@ public void commit()
 
     HadoopUtils.renamePath(this.fs, this.stagingFile, this.outputFile);
 
+    addWriteOutputLineageInfo(this.properties, this.branchId, this.outputFile.toString());
+ }
+
+ protected void addWriteOutputLineageInfo (State state, int branchId, String output) {
+   LineageInfo info = LineageInfo.load(state, LineageInfo.Level.TASK_LEVEL);
+   info.setProp(Joiner.on(".").join(LineageInfo.WRITER_OUTPUT, branchId), output, LineageInfo.Level.TASK_LEVEL);
+   info.save(state);
+
+   if (state.contains(ConfigurationKeys.JOB_DATA_PUBLISHER_TYPE) && state.getProp(ConfigurationKeys.JOB_DATA_PUBLISHER_TYPE).equals(
+       NoopPublisher.class.getName())) {
+     submitter.submit(info.getUUID(), info.getLineageMetaData());
```

Comment:
This seems a bit awkward here. You could refactor this into a separate class, or just specify that Noop publisher will not send lineage events.